src/libstd/collections/hash/table.rs

```diff
@@ -69,50 +67,42 @@ const EMPTY_BUCKET: u64 = 0u64;
 pub struct RawTable<K, V> {
     capacity: usize,
     size:     usize,
-    hashes:   Unique<u64>,
-
-    // Because K/V do not appear directly in any of the types in the struct,
-    // inform rustc that in fact instances of K and V are reachable from here.
-    marker:   marker::PhantomData<(K,V)>,
+    // NB. The table will probably need manual impls of Send and Sync if this
+    // field ever changes.
+    middle: Unique<(K, V)>,
```

Comment:
Note: the `new` code would need to be changed to (following the lead of `Vec`) using a nonzero placeholder pointer.